tools/cli/internal/openapi/oasdiff.go

```diff
@@ -22,16 +22,18 @@ import (
 	"github.com/getkin/kin-openapi/openapi3"
 	"github.com/mongodb/openapi/tools/cli/internal/openapi/errors"
 	"github.com/tufin/oasdiff/diff"
+
 	"github.com/tufin/oasdiff/flatten/allof"
 	"github.com/tufin/oasdiff/load"
 )
 
 type OasDiff struct {
-	base     *load.SpecInfo
-	external *load.SpecInfo
-	config   *diff.Config
-	specDiff *diff.Diff
-	parser   Parser
+	base      *load.SpecInfo
+	external  *load.SpecInfo
+	config    *diff.Config
+	specDiff  *diff.Diff
+	noExtDiff NoExtensionDiff
```

Comment:
The name `NoExtensionDiff` gives fom granted a lot of context on how we use extensions in the spec. Main question from a person without context is what is the difference between `specDiff *diff.Diff` and `noExtDiff NoExtensionDiff`? 

Question: can we have an interface that includes both `specDiff *diff.Diff` and `noExtDiff NoExtensionDiff` as they are both checking differences between specifications

Reply:
starting with decoupling specDiff since we already have an oasdiff result struct in https://github.com/mongodb/openapi/pull/250

Reply:
updated!